pkg/handlers/nutanix/mutation/machinedetails/inject_worker.go

```diff
@@ -0,0 +1,153 @@
+		vars,
+		h.variableName,
+		h.variableFieldPath...,
+	)
+	if err != nil {
+		return err
+	}
+	if !found {
+		log.V(5).Info("Nutanix machine details variable for workers not defined")
+		return nil
+	}
+
+	log = log.WithValues(
+		"variableName",
+		h.variableName,
+		"variableFieldPath",
+		h.variableFieldPath,
+		"variableValue",
+		nutanixMachineDetailsVar,
+	)
+
+	return patches.MutateIfApplicable(
+		obj,
+		vars,
+		&holderRef,
+		selectors.InfrastructureWorkerMachineTemplates(
+			"v1beta1",
+			"NutanixMachineTemplate",
+		),
+		log,
+		func(obj *capxv1.NutanixMachineTemplate) error {
+			log.WithValues(
+				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
+				"patchedObjectName", client.ObjectKeyFromObject(obj),
+			).Info("setting Nutanix machine details in worker NutanixMachineTemplate spec")
+
+			obj.Spec.Template.Spec.BootType = capxv1.NutanixBootType(
+				nutanixMachineDetailsVar.BootType,
+			)
+			obj.Spec.Template.Spec.Cluster = capxv1.NutanixResourceIdentifier{
+				Type: capxv1.NutanixIdentifierType(nutanixMachineDetailsVar.Cluster.Type),
+			}
+			if nutanixMachineDetailsVar.Cluster.Type == v1alpha1.NutanixIdentifierName {
+				obj.Spec.Template.Spec.Cluster.Name = nutanixMachineDetailsVar.Cluster.Name
+			} else {
+				obj.Spec.Template.Spec.Cluster.UUID = nutanixMachineDetailsVar.Cluster.UUID
+			}
+
+			obj.Spec.Template.Spec.Image = capxv1.NutanixResourceIdentifier{
+				Type: capxv1.NutanixIdentifierType(nutanixMachineDetailsVar.Image.Type),
+			}
+			if nutanixMachineDetailsVar.Image.Type == v1alpha1.NutanixIdentifierName {
+				obj.Spec.Template.Spec.Image.Name = nutanixMachineDetailsVar.Image.Name
+			} else {
+				obj.Spec.Template.Spec.Image.UUID = nutanixMachineDetailsVar.Image.UUID
+			}
+
+			obj.Spec.Template.Spec.VCPUSockets = nutanixMachineDetailsVar.VCPUSockets
+			obj.Spec.Template.Spec.VCPUsPerSocket = nutanixMachineDetailsVar.VCPUsPerSocket
+			obj.Spec.Template.Spec.MemorySize = resource.MustParse(
+				nutanixMachineDetailsVar.MemorySize,
+			)
+			obj.Spec.Template.Spec.SystemDiskSize = resource.MustParse(
+				nutanixMachineDetailsVar.SystemDiskSize,
+			)
+			obj.Spec.Template.Spec.Subnets = make([]capxv1.NutanixResourceIdentifier, 0)
+			for _, subnetCRE := range nutanixMachineDetailsVar.Subnets {
+				if subnetCRE.Type == "" {
+					continue
+				}
+				subnet := capxv1.NutanixResourceIdentifier{}
+				if subnetCRE.Type == v1alpha1.NutanixIdentifierName {
+					subnet.Type = capxv1.NutanixIdentifierName
+					if subnetCRE.Name == nil || *subnetCRE.Name == "" {
+						continue
+					}
+					subnet.Name = subnetCRE.Name
+				} else {
+					subnet.Type = capxv1.NutanixIdentifierUUID
+					if subnetCRE.UUID == nil || *subnetCRE.UUID == "" {
+						continue
+					}
+					subnet.UUID = subnetCRE.UUID
+				}
+				obj.Spec.Template.Spec.Subnets = append(obj.Spec.Template.Spec.Subnets, subnet)
+			}
```

Comment:
This will be simplified when we reuse the same types, but lets avoid using "CRE" in the types (since this is already outdated after the rename)
Maybe something like this?
```suggestion
			for _, subnet := range nutanixMachineDetailsVar.Subnets {
				if subnet.Type == "" {
					continue
				}
				subnetIdentifier := capxv1.NutanixResourceIdentifier{}
				if subnet.Type == v1alpha1.NutanixIdentifierName {
					subnetIdentifier.Type = capxv1.NutanixIdentifierName
					if subnet.Name == nil || *subnet.Name == "" {
						continue
					}
					subnetIdentifier.Name = subnet.Name
				} else {
					subnetIdentifier.Type = capxv1.NutanixIdentifierUUID
					if subnet.UUID == nil || *subnet.UUID == "" {
						continue
					}
					subnetIdentifier.UUID = subnet.UUID
				}
				obj.Spec.Template.Spec.Subnets = append(spec.Subnets, subnetIdentifier)
			}
```